cmd/common/oci.go

```diff
@@ -256,12 +261,57 @@ func NewRunCommand(rootCmd *cobra.Command, runtime runtime.Runtime, hiddenColumn
 
 		timeoutDuration := time.Duration(timeoutSeconds) * time.Second
 
-		imageName := args[0]
+		var imageName string
+		if len(args) > 0 {
+			imageName = args[0]
+		}
+
+		paramValueMap := make(map[string]string)
+
+		if inFile != "" {
+			f, err := os.Open(inFile)
+			if err != nil {
+				return fmt.Errorf("opening gadget runtime configuration file %s: %w", inFile, err)
+			}
+			specs, err := gadgetconfig.InstanceSpecsFromReader(f)
+			f.Close()
+			if err != nil {
+				return fmt.Errorf("reading gadget runtime configuration file %s: %w", inFile, err)
+			}
+
+			detachedParam := runtimeParams.Get("detach")
+			isDetach := detachedParam != nil && detachedParam.AsBool()
+
+			if len(specs) > 1 && !isDetach {
+				return fmt.Errorf("multiple gadget runtime configurations found in file %s: this is only supported in combination with --detach", inFile)
+			}
```

Comment:
nit: move this after the `if isDetach` conditional? (It seems to me it breaks a little bit the flow of the logic in the current position).

If it's moved below, then it could just be `if len(specs) > 0` as the isDetach branch was handled before.